var/spack/repos/builtin/packages/sqlite/package.py

```diff
@@ -75,3 +88,11 @@ def configure_args(self):
             args.append('--build=powerpc64le-redhat-linux-gnu')
 
         return args
+
+    @run_after('install')
+    def build_libsqlitefunctions(self):
+        if '+functions' in self.spec:
+            gcc = which('gcc')
+            gcc('-fPIC', '-lm', '-shared', 'extension-functions.c', '-o',
+                'libsqlitefunctions.so')
+            install('libsqlitefunctions.so', self.prefix.lib)
```

Comment:
Use `dso_suffix` instead of `.so` (or `so`, can't remember if it comes with a period) to make this work on macOS.

Reply:
Although neither of us will remember it: It does come without a `.`